### PR TITLE
feat: add verified icon for tips accounts

### PR DESCRIPTION
**Pull request by UncleBill**
https://mask.atlassian.net/browse/MF-527
https://mask.atlassian.net/browse/MF-1134
https://mask.atlassian.net/browse/MF-1135
https://mask.atlassian.net/browse/MF-1113

### CODE REVIEW

File: packages/mask/src/plugins/Tips/components/TipButton.tsx [hunk 1]
```diff
@@ -100,19 +99,19 @@ export const TipButton: FC<Props> = ({
...
     }, [])
 
-    const publicWallets = usePublicWallets(receiver || undefined)
+    const publicWallets = usePublicWallets(personaPubkey)
     const { value: socialAddressList = EMPTY_LIST } = useSocialAddressListAll(visitingIdentity)
     const allAddresses = useMemo(() => {
         switch (pluginId) {
             case NetworkPluginID.PLUGIN_EVM:
-                return uniq([...publicWallets, ...addresses])
+                return uniqBy([...publicWallets, ...addresses], (v) => v.address)
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/components/TipButton.tsx]:
```suggestion
                return uniqBy([...publicWallets, ...addresses], (v) => v.address.toLowerCase())
```

File: packages/mask/src/plugins/Tips/components/TipForm.tsx [hunk 2]
```diff
@@ -1,15 +1,17 @@
+import { CheckIcon, LinkOutIcon, VerifiedIcon } from '@masknet/icons'
 import { useAccount, useChainId, useCurrentWeb3NetworkPluginID, useWeb3State } from '@masknet/plugin-infra/web3'
 import { WalletMessages } from '@masknet/plugin-wallet'
 import { useRemoteControlledDialog } from '@masknet/shared-base-ui'
 import { makeStyles } from '@masknet/theme'
 import { NetworkPluginID } from '@masknet/web3-shared-base'
-import { ChainId as EVM_ChainId } from '@masknet/web3-shared-evm'
+import { ChainId, ChainId as EVM_ChainId } from '@masknet/web3-shared-evm'
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/components/TipForm.tsx]:
```suggestion
import { ChainId } from '@masknet/web3-shared-evm'
```

File: packages/mask/src/plugins/Tips/contexts/Tip/useAddNameService.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    const [map, setMap] = useState<Record<string, string>>({})
+
+    const { NameService } = useWeb3State()
+    const chainId = useChainId()
+    useEffect(() => {
+        if (!NameService?.reverse) return
+        addressConfigs
+            .map((x) => x.address)
+            .forEach(async (addr) => {
+                const name = await NameService.reverse!(chainId, addr)
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/contexts/Tip/useAddNameService.ts]:
```suggestion
                const name = await NameService.reverse?.(chainId, addr)
```

Comment (UncleBill) [reply packages/mask/src/plugins/Tips/contexts/Tip/useAddNameService.ts]: Already check it above, we can assume it's not null.

```ts
        if (!NameService?.reverse) return
```

File: packages/mask/src/plugins/Tips/components/TipButton.tsx (hunk 1 above)
Comment (UncleBill) [reply packages/mask/src/plugins/Tips/components/TipButton.tsx]: done

File: packages/mask/src/plugins/Tips/components/TipForm.tsx (hunk 2 above)
Comment (UncleBill) [reply packages/mask/src/plugins/Tips/components/TipForm.tsx]: done

File: packages/mask/src/plugins/Tips/components/RecipientSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+                        className={cx(classes.link, classes.icon)}
+                        onClick={(e) => e.stopPropagation()}
+                        href={Others?.explorerResolver.addressLink(chainId, addressConfig.address) ?? ''}
+                        target="_blank"
+                        title={t.view_on_explorer()}
+                        rel="noopener noreferrer">
+                        <LinkOutIcon className={classes.actionIcon} />
+                    </Link>
+                    {addressConfig.verified ? <VerifiedIcon className={cx(classes.actionIcon, classes.icon)} /> : null}
+                    {addressConfig.address === recipient ? (
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/components/RecipientSelect.tsx]:
```suggestion
                    {isSameAddress(addressConfig.address, recipient)? (
```

Comment (UncleBill) [reply packages/mask/src/plugins/Tips/components/RecipientSelect.tsx]: done

File: packages/mask/src/plugins/Tips/components/TipButton.tsx [hunk 5]
```diff
@@ -4,22 +4,24 @@ import { useCurrentWeb3NetworkPluginID, useSocialAddressListAll } from '@masknet
...
+import { useProfilePublicKey } from '../hooks/useProfilePublicKey'
 import { usePublicWallets } from '../hooks/usePublicWallets'
 import { useI18N } from '../locales'
 import { PluginNextIDMessages } from '../messages'
+import type { AddressConfig } from '../types'
 
 interface Props extends HTMLProps<HTMLDivElement> {
-    addresses?: string[]
+    addresses?: AddressConfig[]
+    recipient?: AddressConfig['address']
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/components/TipButton.tsx]:
```suggestion
    recipient?: string
```

File: packages/mask/src/plugins/Tips/contexts/Tip/useAddNameService.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import { useChainId, useWeb3State } from '@masknet/plugin-infra/web3'
+import { useEffect, useMemo, useState } from 'react'
+import type { AddressConfig } from '../../types'
+
+export function useAddNameService(addressConfigs: AddressConfig[]) {
+    const [map, setMap] = useState<Record<string, string>>({})
+
+    const { NameService } = useWeb3State()
+    const chainId = useChainId()
+    useEffect(() => {
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/contexts/Tip/useAddNameService.ts]: The total hook can refactor to use `useAsync`.

```ts
return useAsync(async () => {
    const names = await getNames()
    
    // compose result
    return configs.map()
}, [])
```

File: packages/mask/src/plugins/Tips/contexts/Tip/useAddNameService.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
+import { useChainId, useWeb3State } from '@masknet/plugin-infra/web3'
+import { useEffect, useMemo, useState } from 'react'
+import type { AddressConfig } from '../../types'
+
+export function useAddNameService(addressConfigs: AddressConfig[]) {
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/contexts/Tip/useAddNameService.ts]: Use a proper name.

File: packages/mask/src/plugins/Tips/hooks/usePublicWallets.ts [hunk 8]
```diff
@@ -35,15 +36,15 @@ export function usePublicWallets(profile: ProfileIdentifier | undefined) {
         if (!tipWallets) return null
         return tipWallets
             .filter((x) => {
-                if (NextIDWalletsState.value) {
+                if (nextIdWallets) {
                     // Sometimes, the wallet might get deleted from next.id
-                    return x.isPublic && NextIDWalletsState.value.includes(x.identity)
+                    return x.isPublic && nextIdWallets.find((y) => y.address === x.identity)
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/hooks/usePublicWallets.ts]: If `x.identity` is an address, please use `isSameAddress`.

File: packages/mask/src/plugins/Tips/types/tip.ts [hunk 9]
```diff
@@ -3,10 +3,15 @@ export enum TipType {
     NFT = 'nft',
 }
 
+export interface AddressConfig {
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/types/tip.ts]: Only a bit of advice.

```suggestion
export interface TipAccount {
```

File: packages/mask/src/plugins/Tips/components/TipButton.tsx (hunk 5 above)
Comment (UncleBill) [reply packages/mask/src/plugins/Tips/components/TipButton.tsx]: I prefer to reuse typing. When `address` in `AddressConfig` gets updated, no need to update this prop as well.

File: packages/mask/src/social-network-adaptor/twitter.com/collecting/identity.ts [hunk 10]
```diff
@@ -116,14 +117,26 @@ function resolveLastRecognizedIdentityMobileInner(
...
+    return first(slugs)
+}
+
 function resolveCurrentVisitingIdentityInner(
     ref: Next.CollectingCapabilities.IdentityResolveProvider['recognized'],
     cancel: AbortSignal,
 ) {
     const avatarSelector = searchAvatarSelector()
     const avatarMetaSelector = searchAvatarMetaSelector()
+    let firstSlug = getFirstSlug()
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/twitter.com/collecting/identity.ts]: Please refactor it in #6615. Learn more https://mask.atlassian.net/browse/MF-1256.

File: packages/mask/src/plugins/Tips/contexts/Tip/useAddNameService.ts (hunk 6 above)
Comment (UncleBill) [reply packages/mask/src/plugins/Tips/contexts/Tip/useAddNameService.ts]: `async-await` is not in the top level of `useEffect` callback

File: packages/mask/src/plugins/Tips/contexts/Tip/useAddNameService.ts (hunk 7 above)
Comment (UncleBill) [reply packages/mask/src/plugins/Tips/contexts/Tip/useAddNameService.ts]: Any suggestion?

File: packages/mask/src/plugins/Tips/hooks/usePublicWallets.ts (hunk 8 above)
Comment (UncleBill) [reply packages/mask/src/plugins/Tips/hooks/usePublicWallets.ts]: Fixed.

File: packages/mask/src/plugins/Tips/types/tip.ts (hunk 9 above)
Comment (UncleBill) [reply packages/mask/src/plugins/Tips/types/tip.ts]: Good suggestion. Changed.

File: packages/mask/src/plugins/Tips/contexts/Tip/useAddNameService.ts [hunk 11]
```diff
@@ -0,0 +1,29 @@
+import { useChainId, useWeb3State } from '@masknet/plugin-infra/web3'
+import { useEffect, useMemo, useState } from 'react'
+import type { TipAccount } from '../../types'
+
+export function useAddNameService(tipAccounts: TipAccount[]) {
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/contexts/Tip/useAddNameService.ts]:
```suggestion
export function useTipAccountsComputed(tipAccounts: TipAccount[]) {
```

File: packages/mask/src/plugins/Tips/contexts/Tip/useAddNameService.ts (hunk 6 above)
Comment (guanbinrui) [reply packages/mask/src/plugins/Tips/contexts/Tip/useAddNameService.ts]: So, we can use `await Promise.all()`.

**Pull request merged**